[Accessibility] Convert the remove item link into a button on cart and mini-cart

Manussakis opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

<!-- For bug fixes: If known, please provide links to help with traceability and escape analysis. -->
<!-- Please include a link to the issue of the bug being fixed, if one doesn't exist please create it. -->
<!-- If the PR that introduced the bug is known, please also add its link below. -->

Closes #56606 .

The "Remove from cart" button on the legacy Minicart is not actually a button, but a link. It uses the `<a>` HTML tag.
This is a semantics issue because a link should be used to link to another page or a page section. For the action it does, the `<button>` is the appropriate HTML tag.
However, changing the HTML is problematic because it's prone to regression, so the `role="button"` attribute was added to the `<a>` tag.
Once the link has a button role, it also needs to act as a button. Which means that it should be possible to click on it using the keyboard Space key. However, it's very unlikely that users click on it using the Space key since the legacy minicart can only be opened once the user hovers over the cart icon with the mouse. But we wanted to cover all cases and implemented the feature anyway.


(For Bug Fixes) Bug introduced in PR # .

### Screenshots or screen recordings:

<!-- If this PR includes UI changes, please provide screenshots or a screen recording for clarity. -->
<!-- This section can be removed if not applicable. -->

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

This PR should also be tested with Storefront theme, but I will detail the steps to QA using the Astra theme as it was flagged on the issue.

Here's how to insert the WooCommerce Cart in the header in Astra.

1. Install the Astra theme.
1. Go to Appearance > Customize.
2. Click Header Builder.
3. In the header builder section that opens at the bottom of the main content in the Customizer, hover over a column to 4. see a + icon appear.
5. Click on the column and an "Insert Elements" modal appears.
6. In the insert elements modal, click on the cart. This will insert the cart in the header.
7. Publish changes.
![image](https://github.com/user-attachments/assets/d85768c0-135e-41fc-9e96-ea5a3853e894)

The steps below should be done with the WooCommerce -> Settings -> Products -> Add to cart behaviour -> "Enable AJAX add to cart buttons on archives" option both disabled and enabled.

8. Go to the shop page on the frontend.
9. Add a product to the cart.
10. Hover over the Mini-cart on the header.
11. Verify the remove item link has the `role="button"` attribute. 💡**Tip**: The Mini-cart content only appears while the mouse is over it, so you may need to open the Dev tools with the shorthand command. (Command + Shif + c, on Mac).
![Screenshot 2025-03-31 at 16 43 03](https://github.com/user-attachments/assets/473cbf46-c0db-499c-bb73-2954970dff66)

12. Verify the remove item link can be clicked by pressing the Space key on the keyboard. Once the Mini-cart appears, click on the subtotal text and press Shift + Tab until you get to the remove item link. Then, press the Space key on the keyboard.
13. Go to the Cart page.
14. Verify the remove item link has the `role="button"` attribute.
![image](https://github.com/user-attachments/assets/68d2558c-e64d-4ed4-b7fb-b1905b60983e)

15. Turn on the screen reader.
16. Focus on the remove item link.
17. Verify it's announced as a button.
18. Verify the link can be clicked by pressing the Space key on the keyboard.

<!-- End testing instructions -->

### Testing that has already taken place:

<!-- Detail any testing that has already been conducted. -->
<!-- Include environment details such as hosting type, plugins, theme, store size, store age, and relevant settings. -->
<!-- Mention any analysis performed, such as assessing potential impacts on environment attributes and other plugins, performance profiling, or LLM/AI-based analysis. -->
<!-- Within the testing details you provide, please ensure that no sensitive information (such as API keys, passwords, user data, etc.) is included in this public pull request. -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **New Features**
	- Improved keyboard accessibility for removing items from the cart and mini-cart, allowing users to use the Space key to activate the remove action.
- **Style**
	- Updated remove item links in the cart and mini-cart to use button semantics for better accessibility.
- **Tests**
	- Added end-to-end tests to verify keyboard accessibility for removing items from the mini-cart.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Manussakis committed:
Convert the remove item link into a button on cart and mini-cart

Manussakis committed:
Add changelog file

Manussakis committed:
Update cart and mini-cart templates version

Manussakis committed:
Trigger remove item click when the Space key is pressed

Manussakis commented:
@dkotter This PR is ready for code review.
You will notice that some tests are failing, but I couldn't associate them with the changes made. I also noticed that the same tests are failing on another PR I am working on (https://github.com/woocommerce/woocommerce/pull/56875), so I assume that this is on trunk.

dkotter reviewed (dismissed):
@ankitguptaindia @qasumitbagthariya This is ready for QA

ankitguptaindia commented:
@Manussakis Thanks for the fixes. 

 - ⚠️  - I can see the "role="button" attribute on the mini cart when it is open in hover, but it doesn't click/remove when I press the space bar but it scrolls the page instead.
- ✅ - Changes on the Cart page work as expected. 


![Screenshot 2025-04-01 at 11 21 46 AM](https://github.com/user-attachments/assets/928aa8a1-8b65-414a-9bc7-d9931c0dc88c)



**Please have a look in below screencast for my steps:**

https://github.com/user-attachments/assets/9ad2438b-6474-4714-ab6a-39cc01397dc5

Manussakis commented:
Thanks, @ankitguptaindia for testing this PR 🙂

I was not able to reproduce the issue you flagged, please watch the video below.
Could you please tell me more about your testing environment?
Thanks!

https://github.com/user-attachments/assets/6f5a4bdc-da2d-4990-9f3f-762dec75d44f

ankitguptaindia commented:
Hey @Manussakis Thanks for looking into it.

Regarding your query:
>Could you please tell me more about your testing environment? 

To ensure the issue is not environment-specific, I tested it on different setups and devices, and I experienced the same issue as in my usual testing environment.

I debugged this in detail and observed that the issue occurs when the store has the "Enable AJAX add to cart buttons on archives" setting disabled on the Product Settings page.

When this setting is enabled, the product removal icon works as described in the PR.

![image](https://github.com/user-attachments/assets/f023cc66-bc6b-4e38-ac06-c9b511fcaaa2)

**Please have a look in this video for more details:** 

https://github.com/user-attachments/assets/8e50b086-3b55-444e-a0ca-3a63c2ab48b9




----


**Next Step-** Moving back for remediation.

Manussakis committed:
Merge branch 'trunk' into fix/56606

Manussakis committed:
Trigger remove item click when the Space key is pressed with AJAX disabled

Manussakis commented:
Thanks, @ankitguptaindia for debugging it!

@dkotter Please review this code update:
[Trigger remove item click when the Space key is pressed with AJAX dis…](https://github.com/woocommerce/woocommerce/pull/56869/commits/e63a7db92d77d072a090420f261b675bec6fdc27)

Thanks!

dkotter commented on plugins/woocommerce/client/legacy/js/frontend/woocommerce.js:
```diff
@@ -182,9 +182,27 @@ jQuery( function ( $ ) {
 		} );
 	} );
 
+	// If the the "Enable AJAX add to cart buttons on archives" setting is disabled
```
```suggestion
	// If the "Enable AJAX add to cart buttons on archives" setting is disabled
```

dkotter commented on plugins/woocommerce/client/legacy/js/frontend/woocommerce.js:
```diff
@@ -182,9 +182,27 @@ jQuery( function ( $ ) {
 
+	// If the the "Enable AJAX add to cart buttons on archives" setting is disabled
+	// the add-to-cart.js file won't be loaded, so we need to add the event listener here.
+	if( typeof wc_add_to_cart_params === 'undefined') {
```
```suggestion
	if ( typeof wc_add_to_cart_params === 'undefined') {
```

dkotter commented on plugins/woocommerce/client/legacy/js/frontend/woocommerce.js:
```diff
@@ -182,9 +182,27 @@ jQuery( function ( $ ) {
 
+/**
+ * Handle when pressing the Space key on the remove item link.
+ * This is necessary because the link got the role="button" attribute
```
```suggestion
 * This is necessary because the link has the role="button" attribute
```

ankitguptaindia commented:
**QA/Test Report-** ✅ 
------------

**Testing Environment -** 

<Details>

* WordPress: 6.7.2
* Theme:  
 - Astra
* WooCommerce - Development version with this fix-specific branch.  
* PHP: 8.3
* Web Server: Nginx
* Browser: Chrome - Version 134
* OS: macOS Sonoma 14.6.1 


</Details>


**Test Results -** 

The changes work as intended and have been tested across Safari, Firefox, and Chrome. 

**Functional Demo /  Screencast -**


https://github.com/user-attachments/assets/b3a028d4-a5fc-4768-bd94-3a1a91cfa0d1

ankitguptaindia commented:
@Manussakis Could you please open this PR.

Manussakis committed:
Remove unnecessary word in an inline comment

Co-authored-by: Darin Kotter <darin.kotter@gmail.com>

Manussakis committed:
Fix comment grammar

Co-authored-by: Darin Kotter <darin.kotter@gmail.com>

Manussakis committed:
Format if statement

Co-authored-by: Darin Kotter <darin.kotter@gmail.com>

nielslange commented on plugins/woocommerce/client/legacy/js/frontend/add-to-cart.js:
```diff
@@ -169,6 +170,18 @@ jQuery( function( $ ) {
+			e.preventDefault();
+			$( this ).trigger( 'click' );
+		}
+	};
```
For increased readability, I suggest using `event` instead of `e`. It's something we also do in other functions.

```suggestion
	AddToCartHandler.prototype.onKeydownRemoveFromCart = function( event ) {
		if ( event.key === ' ' ) {
			event.preventDefault();
			$( this ).trigger( 'click' );
		}
	};
```

nielslange commented on plugins/woocommerce/client/legacy/js/frontend/cart.js:
```diff
@@ -761,6 +766,20 @@ jQuery( function ( $ ) {
+				evt.preventDefault();
+				$( evt.currentTarget ).trigger( 'click' );
+			}
+		},
```
Similar as before, changing `evt` to `event` would increase the readability here:

```suggestion
		on_keydown_remove_item: function ( event ) {
			if ( event.key === ' ' ) {
				event.preventDefault();
				$( event.currentTarget ).trigger( 'click' );
			}
		},
```

nielslange commented on plugins/woocommerce/client/legacy/js/frontend/woocommerce.js:
```diff
@@ -182,9 +182,27 @@ jQuery( function ( $ ) {
+		e.preventDefault();
+		e.currentTarget.click();
+	}
+}
```
Finally, in this case, we could also use `event` over `e`:

```suggestion
function on_keydown_remove_from_cart( event ) {
	if ( event.key === ' ' ) {
		event.preventDefault();
		event.currentTarget.click();
	}
}
```

nielslange reviewed (commented):
Thanks for working on this, @Manussakis. I left a few suggestions to improve readability.

Quick question on step 12 of the testing instructions:

> _Verify the remove item link can be clicked by pressing the Space key on the keyboard. Once the Mini-cart appears, click on the subtotal text and press Shift + Tab until you get to the remove item link. Then, press the Space key on the keyboard._

I’m unable to open the Mini Cart using just the keyboard. Is this working for you?

Manussakis commented:
@nielslange 

> I’m unable to open the Mini Cart using just the keyboard. Is this working for you?

If you are testing using Storefront or Astra themes, this is not possible, it only appears when hovering over it since clicking on it leads to the Cart page. That's why I left this tip:

> Once the Mini-cart appears, click on the subtotal text and press Shift + Tab until you get to the remove item link. Then, press the Space key on the keyboard.

Please let me know if that helps 🙂

Manussakis committed:
Use event instead of e

Co-authored-by: Niels Lange <codenex@gmail.com>

Manussakis committed:
Use event instead of evt

Co-authored-by: Niels Lange <codenex@gmail.com>

Manussakis committed:
Use event instead of e

Co-authored-by: Niels Lange <codenex@gmail.com>

Manussakis committed:
Merge branch 'trunk' into fix/56606

nielslange commented:
> @nielslange
> 
> > I’m unable to open the Mini Cart using just the keyboard. Is this working for you?
> 
> If you are testing using Storefront or Astra themes, this is not possible, it only appears when hovering over it since clicking on it leads to the Cart page. That's why I left this tip:
> 
> > Once the Mini-cart appears, click on the subtotal text and press Shift + Tab until you get to the remove item link. Then, press the Space key on the keyboard.
> 
> Please let me know if that helps 🙂

My question was a bit more general, @Manussakis. I’m wondering why we highlight the Space key interaction in the testing steps. Since the Mini Cart isn’t easily accessible via keyboard in most themes, it’s unlikely that users would interact with it this way, making the scenario feel somewhat abstract.

I also noticed the addition of keyboard handling logic like `.on('keydown', '.remove_from_cart_button', this.onKeydownRemoveFromCart)`, but since this part of the UI is only accessible via mouse, the keyboard support doesn't feel particularly actionable in practice.

To ensure this change works as expected, and continues to do so, could you add an integration test for the changes?

nielslange commented:
> @nielslange just noting here that @Manussakis will be looking into this next week when he is back on the project.

Thanks for letting me know, @vikrampm1. As tomorrow will be my last working day at Automattic, I've unassigned myself as a reviewer and requested a review from @woocommerce/rubik.

Manussakis committed:
Merge branch 'trunk' into fix/56606

Manussakis committed:
Merge branch 'trunk' into fix/56606

Manussakis commented:
@woocommerce/rubik 

Replying to [this concern](https://github.com/woocommerce/woocommerce/pull/56869#issuecomment-2820597048) from @nielslange 

The "Remove from cart" button on the legacy Mini Cart is not actually a button, but a link. It uses the `<a>` HTML tag.
This is a semantics issue because a link should be used to link to another page or a page section. For the action it does, the `<button>` is the appropriate HTML tag.
However, changing the HTML is problematic because it's prone to regression, so the `role="button"` attribute was added to the `<a>` tag.
Once the link has a button role, it also needs to act as a button. Which means that it should be possible to click on it using the keyboard Space key. However, it's very unlikely that users click on it using the Space key since the legacy Mini Cart can only be opened once the user hovers over the cart icon with the mouse. But we wanted to cover all cases and implemented the feature anyway.

@nielslange suggested that an integration test should be done, but I didn't see any example of it for legacy templates like the Mini Cart. I only see tests for the Mini Cart block, which was not touched in this PR.

@woocommerce/rubik could you assess the PR and confirm an integration test is needed? If so, could you please provide me with any guidance on that?

Thanks!

@vikrampm1

Manussakis committed:
Merge branch 'trunk' into fix/56606

mikejolley commented:
@Manussakis I agree with change to role button.

Does the `<a role="button">` also need tabindex? And should it be space and enter keyboard events? We can use keycodes like in WP core https://github.com/WordPress/WordPress/blob/779dc9c93d18c19c41897d5b993c77eec6787edc/wp-admin/js/application-passwords.js#L153C2-L158C6

R/e tests. Legacy cart does not have an integration test. You could add something to the e2e tests here:
https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/e2e-core-tests/src/specs/shopper/front-end-cart.test.js#L66

Manussakis committed:
Merge branch 'trunk' into fix/56606

faisal-alvi committed:
Add test to remove the item from the min-cart

faisal-alvi committed:
fix eslint issues,

faisal-alvi committed:
Merge branch 'trunk' into fix/56606

faisal-alvi committed:
a few more eslint fixes

/home/runner/work/woocommerce/woocommerce/packages/js/e2e-core-tests/src/specs/shopper/front-end-cart.test.js
  84:57  error  Replace `el·=>·el.getAttribute(·'role'·)·` with `(·el·)·=>⏎↹↹↹↹el.getAttribute(·'role'·)⏎↹↹↹`                   prettier/prettier
  91:40  error  Replace `·'.woocommerce-mini-cart__empty-message',·` with `⏎↹↹↹↹'.woocommerce-mini-cart__empty-message',⏎↹↹↹↹`  prettier/prettier
  92:1   error  Insert `↹`                                                                                                      prettier/prettier
  93:4   error  Replace `}·` with `↹}⏎↹↹↹`                                                                                      prettier/prettier

✖ 4 problems (4 errors, 0 warnings)
  4 errors and 0 warnings potentially fixable with the `--fix` option.

mikejolley commented on plugins/woocommerce/templates/cart/cart.php:
```diff
@@ -12,7 +12,7 @@
  * @see     https://woocommerce.com/document/template-structure/
  * @package WooCommerce\Templates
- * @version 7.9.0
+ * @version 9.9.0
```
Needs bump to 10.0.0

amesplant replied:
@mikejolley this is now bumped to 10.0.0

mikejolley commented on plugins/woocommerce/templates/cart/mini-cart.php:
```diff
@@ -14,7 +14,7 @@
  * @see     https://woocommerce.com/document/template-structure/
  * @package WooCommerce\Templates
- * @version 9.4.0
+ * @version 9.9.0
```
Needs bump to 10.0.0

amesplant replied:
@mikejolley this is now bumped to 10.0.0

mikejolley reviewed (dismissed):
I've tested functionality—happy with this, it looks good to go. There are 2 changes needed before this can merge.

1. 2 template bumps need to increase to 10.0
2. @faisal-alvi The test added to `core-e2e-tests` won't run. That is apparently deprecated. Could you add the test under `tests/e2e-pw/tests/cart` instead? `e2e-pw` will run on the CI.

Thanks!

opr commented:
@vikrampm1 please see above, there are outstanding suggestions from Mike that need to be addressed before this can be merged.

https://github.com/woocommerce/woocommerce/pull/56869#pullrequestreview-2881591212

vikrampm1 commented:
@opr got it. We will address those asap and ping you back.

amesplant committed:
Bump cart template to version 10.0.0

amesplant committed:
Update mini-cart template to version 10.0.0

amesplant committed:
Merge branch 'trunk' into fix/56606

amesplant committed:
Move test to remove the item from the min-cart into e2e-pw

opr commented on plugins/woocommerce/tests/e2e-pw/tests/cart/add-to-cart.spec.js:
```diff
@@ -9,6 +9,8 @@ import { addAProductToCart } from '@woocommerce/e2e-utils-playwright';
 import { tags, test } from '../../fixtures/fixtures';
 import { WC_API_PATH } from '../../utils/api-client';
 import { checkCartContentInBlocksCart } from '../../utils/cart';
+import { it } from '@jest/globals';
```
This is not used

opr commented on plugins/woocommerce/tests/e2e-pw/tests/cart/add-to-cart.spec.js:
```diff
@@ -9,6 +9,8 @@ import { addAProductToCart } from '@woocommerce/e2e-utils-playwright';
 import { WC_API_PATH } from '../../utils/api-client';
 import { checkCartContentInBlocksCart } from '../../utils/cart';
+import { it } from '@jest/globals';
+import { shopper, uiUnblocked } from '@woocommerce/e2e-utils';
```
This needs to be moved to the

```
/**
 * External dependencies
 */
```

block

opr reviewed (changes requested):
The E2E test added seems to be for a deprecated test setup we had and transferring it over directly like this doesn't seem to work. We should rewrite the test to work with the current system.

Some instructions to help get the e2e suite running locally:

- `pnpm --filter=@woocommerce/plugin-woocommerce env:start`
- `pnpm --filter=@woocommerce/plugin-woocommerce test:e2e --ui`

You won't see the `add-to-cart-spec` in the list because there's an error with the newly added test (`page` doesn't exist).

I bet if you give an agent the tests in `client/blocks/test/e2e` as context, it would be able to rewrite this one easily.

faisal-alvi committed:
update the e2e test

faisal-alvi committed:
Update e2e test mini cart button

faisal-alvi committed:
suggestions implemented

opr commented:
@faisal-alvi I see you pushed changes, are there more to come? CI is still failing. The test should use updated e2e utils from `@woocommerce/e2e-utils-playwright`. An example of a test that adds an item to the cart is here https://github.com/woocommerce/woocommerce/blob/f2b16512c421c90131e1e0afac1f36bc6c455113/plugins/woocommerce/tests/e2e-pw/tests/coupons/cart-checkout-coupons.spec.js#L234

faisal-alvi committed:
update mini cart removal test to use test steps for better clarity and structure

faisal-alvi committed:
update e2e to use the block based min-cart selectors

faisal-alvi committed:
click instead of hover

faisal-alvi committed:
fix linting issues

faisal-alvi committed:
Merge branch 'trunk' into fix/56606

faisal-alvi committed:
a couple more linting fixes,
. lint:changes:branch:js:   119:20  error    Replace `·'.wc-block-mini-cart__empty-cart-wrapper'·` with `⏎↹↹↹↹↹↹↹'.wc-block-mini-cart__empty-cart-wrapper'⏎↹↹↹↹↹↹`  prettier/prettier
. lint:changes:branch:js:   122:20  error    Replace `·'.wc-block-mini-cart__empty-cart-wrapper'·` with `⏎↹↹↹↹↹↹↹'.wc-block-mini-cart__empty-cart-wrapper'⏎↹↹↹↹↹↹`  prettier/prettier

faisal-alvi committed:
Merge branch 'fix/56606' of https://github.com/Manussakis/woocommerce into fix/56606

faisal-alvi commented:
@opr Thanks for the help! I’ve implemented the suggestions and updated the E2E test - it’s now passing. However, another E2E test is failing: `tests/e2e/tests/checkout/checkout-block.shopper.block_theme.spec.ts:173:2`

This failure seems unrelated to the changes in this PR and I don’t see an option to re-run the tests on my end to check if it’s a flaky one. Could you please re-run the tests or let me know if you have any suggestions?

opr commented:
The tests are passing, the failure on the job is probably because this is on a fork. The check passes locally for me.

```
pnpm analyzer "03d7702625dcff5e35ed372bb787eba5fb532da0" $version --ba
 ✔  08:43:12 p.m. 
se "6dfd5f16aecbeee2fae0ec30e0c7ce7036cfeaac" --outputStyle "github" --exclude "plugins/woocommerce/tests"

> code-analyzer@1.0.0 analyzer /Users/thomas/code/tomato/app/public/wp-content/plugins/woocommerce-monorepo/tools/code-analyzer
> node -r ts-node/register ./src/commands/analyzer/index.ts "03d7702625dcff5e35ed372bb787eba5fb532da0" "3.7.1" "--base" "6dfd5f16aecbeee2fae0ec30e0c7ce7036cfeaac" "--outputStyle" "github" "--exclude" "plugins/woocommerce/tests"

✔ Making temporary clone of /Users/thomas/code/tomato/app/public/wp-content/plugins/woocommerce-monorepo/...... complete.
Temporary clone of /Users/thomas/code/tomato/app/public/wp-content/plugins/woocommerce-monorepo/ created at /var/folders/pn/bdk4xh7x4t9cf18mzm8d5r6c0000gn/T/code-analyzer-tmp/e62d055f-3acc-46c3-9a54-bb3a4209de35
✔ Detecting hook changes...... complete.
⠦ Detecting template changes......Unable to get PR number in /plugins/woocommerce/templates/cart/cart.php:15
⠸ Detecting template changes......Unable to get PR number in /plugins/woocommerce/templates/cart/cart.php:63
⠦ Detecting template changes......Unable to get PR number in /plugins/woocommerce/templates/cart/mini-cart.php:17
⠋ Detecting template changes......Unable to get PR number in /plugins/woocommerce/templates/cart/mini-cart.php:50
✔ Detecting template changes...... complete.
✔ Detecting DB changes...... complete.
::warning file=/plugins/woocommerce/templates/cart/cart.php,line=1,title=TEMPLATES::This template may require a version bump! Expected 3.7.1
::warning file=/plugins/woocommerce/templates/cart/mini-cart.php,line=1,title=TEMPLATES::This template may require a version bump! Expected 3.7.1

::set-output name=templates::\n\n### Template changes:\n* **File:** /plugins/woocommerce/templates/cart/cart.php\n  * WARNING: This template may require a version bump! Expected 3.7.1\n* **File:** /plugins/woocommerce/templates/cart/mini-cart.php\n  * WARNING: This template may require a version bump! Expected 3.7.1


## HOOK CHANGES
---------------------------------------------------
No hook changes found.
---------------------------------------------------


## DB CHANGES
---------------------------------------------------
No db changes found.
---------------------------------------------------
```

faisal-alvi committed:
Merge branch 'trunk' into fix/56606

opr reviewed (approved):
Thanks, tests are passing, the failing CI check is because this is a fork. Approving! 👏🏼

opr committed:
Merge branch 'trunk' into fix/56606

vikrampm1 commented:
Thanks @opr please proceed and merge the PR if there are no concerns here. Hopefully this makes it to the 10.0 before the code freeze.
